stac_pydantic/links.py

```diff
@@ -15,7 +15,7 @@ class Link(StacBaseModel):
 
     href: str = Field(..., alias="href", min_length=1)
     rel: str = Field(..., alias="rel", min_length=1)
-    type: Optional[MimeTypes] = None
+    type: Optional[str] = None
```

Comment:
as mentioned in https://github.com/stac-utils/stac-pydantic/issues/181#issuecomment-2944119599 could me make this an union between `MimeTypes` and `str`?